transport-protocol/src/custom.rs

```diff
@@ -269,6 +277,24 @@ pub fn serialize_proxy_psa_attestation_token(
     Ok(proxy_attestation_server_request.write_to_bytes()?)
 }
 
+pub fn serialize_nitro_attestation_doc(doc: &[u8], device_id: i32) -> TransportProtocolResult {
```

Comment:
I think `doc` is better as a `Vec<u8>` given this function allocates a new vector using `doc.to_vec()`.